Pod/Classes/XExtensionItem.h

```diff
@@ -89,17 +131,74 @@ typedef UIImage *(^XExtensionItemThumbnailProvidingBlock)(CGSize suggestedSize,
 @property (nonatomic) XExtensionItemReferrer *referrer;
 
 /**
- An optional dictionary of keys and values. Individual applications can add advertise whatever custom parameters they
- support, and extension developers can add values for those parameters in this dictionary.
+ A block that will be called with the suggested size for the thumbnail image, in points, and the activity type chosen by 
+ the user. It should return an image using the appropriate scale for the screen. Images provided at the suggested size 
+ will result in the best experience.
+ */
+@property (nonatomic, copy) XExtensionItemThumbnailProvidingBlock thumbnailProvider;
+
+/**
+ Add parameters from a dictionary-serializable custom parameters object.
+ 
+ @param dictionarySerializable A custom parameters object.
+ */
+- (void)addCustomParameters:(id <XExtensionItemCustomParameters>)dictionarySerializable;
+
+/**
+ An optional dictionary of keys and values.
+ 
+ @discussion Please look at `addCustomParameters` and the `XExtensionItemCustomParameters` protocol before considering
+ populating this dictionary directly.
  
- Custom user info keys should *not* start with `x-extension-item`, as those will be used internally by this library.
+ Individual applications can add advertise the custom parameter keys that they support, and extension developers can add 
+ values for those parameters to this dictionary, but a better solution is for third-party developers to create an object 
+ that conforms to `XExtensionItemCustomParameters`, and open a pull request to add it to the `XExtensionItem` 
+ repository. See `XExtensionItemTumblrParameters` for an example of this.
+ 
+ Custom user info keys should *not* start with `x-extension-item`, as those are used internally by this library. Custom 
+ user info keys are also at risk of colliding with keys found in the dictionary representations of custom parameters 
```

Comment:
> Custom user info keys are also at risk of colliding with keys found in the dictionary representations of custom parameters objects added via `addCustomParameters:`.

This is pretty lame. I suppose I could nest the `customParameters` dictionary inside the `userInfo` dictionary? This would make it easier to populate an `XExtensionItemSource` without thinking too much about it, but would require more thought when grabbing the values on the other end.

@mattbischoff @AriX?